modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java

```diff
@@ -264,6 +326,14 @@ private class S3BlobStoreHttpHandler extends S3HttpHandler implements BlobStoreH
 
         @Override
         public void handle(final HttpExchange exchange) throws IOException {
+            final String request = exchange.getRequestMethod() + " " + exchange.getRequestURI();
+            if (shouldFailCompleteMultipartUploadRequest.get() && Regex.simpleMatch("POST /*/*?uploadId=*", request)) {
+                try (exchange) {
+                    drainInputStream(exchange.getRequestBody());
+                    exchange.sendResponseHeaders(RestStatus.INTERNAL_SERVER_ERROR.getStatus(), -1);
```

Comment:
I changed it to randomise between 400 (non-retryable) and 429 (retryable) to give it a bit more coverage since the count should be correct in either case.